### PR TITLE
Annotations branch fix

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Ensure that inputs variables to a branch node have unique names instead of copying the name from the promise as is.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/602

### CODE REVIEW

File: tests/flytekit/unit/annotated/test_serialization.py [hunk 1]
```diff
@@ -97,6 +97,42 @@ def my_wf(a: int, b: str) -> (int, str):
...
+    @task
+    def t2(a: str) -> str:
+        return a
+
+    @workflow
+    def my_wf(a: int, b: str) -> (int, str):
+        x, y = t1(a=a)
+        d = (
+            conditional("test1")
+            .if_(x == 4)
```
Comment (wild-endeavor) [inline tests/flytekit/unit/annotated/test_serialization.py:114]: Can we add two more test cases?  one where the if_ statement has an `||` in it.  and also, one where the then statements call subworkflows?

File: flytekit/annotated/condition.py [hunk 2]
```diff
@@ -269,14 +269,19 @@ def else_(self) -> Case:
 }
 
 
+def create_branch_node_promise_var(node_id: str, var: str) -> str:
+    return f"{node_id}.{var}"
```
Comment (wild-endeavor) [inline flytekit/annotated/condition.py:273]: Could you add a comment explaining why this is necessary?  Is it just in cases where there's an out_0 from two parents?

Comment (EngHabu) [reply flytekit/annotated/condition.py:273]: true... will do

File: flytekit/annotated/condition.py [hunk 3]
```diff
@@ -269,14 +269,26 @@ def else_(self) -> Case:
 }
 
 
+# When building bindings for the branch node, the inputs to the conditions (e.g. (x==5)) need to have variable names
```
Comment (kumare3) [inline flytekit/annotated/condition.py:272]: can we make this a docstring?

File: flytekit/annotated/condition.py [hunk 4]
```diff
@@ -292,7 +304,7 @@ def transform_to_conj_expr(expr: ConjunctionExpression) -> (_core_cond.Conjuncti
 
 def transform_to_operand(v: Union[Promise, Literal]) -> (_core_cond.Operand, Optional[Promise]):
     if isinstance(v, Promise):
-        return _core_cond.Operand(var=v.var), v
+        return _core_cond.Operand(var=create_branch_node_promise_var(v.ref.node_id, v.var)), v
```
Comment (kumare3) [inline flytekit/annotated/condition.py:307]: so indeed this was the only change needed on flytekit side?

**Comment by kumare3** [comment]
lgtm

**Review by kumare3 — APPROVED** [review]

**Pull request merged**